extensions/wikia/Our404Handler/SpecialOur404Handler_body.php

```diff
@@ -83,7 +83,9 @@ public function doRender404() {
 		if( !is_null( $oTitle ) ) {
 			// Preserve the query string on a redirect
 			$query = parse_url ( $_SERVER['REQUEST_URI'], PHP_URL_QUERY );
-			if( $namespace == NS_SPECIAL || $namespace == NS_MEDIA ) {
+			if( $namespace == NS_SPECIAL || $namespace == NS_MEDIA
+				|| ( $namespace && $wgOur404HandlerBroaderRedirects )
```

Comment:
I'm not sure what this check is exactly for but please remember that NS_MAIN namespace is actually integer 0 which in this comparison would be "false" (vs. "true" for all other namespaces).